pkg/plugins/runtime/k8s/controllers/configmap_controller.go

```diff
@@ -0,0 +1,137 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	kube_core "k8s.io/api/core/v1"
+	kube_runtime "k8s.io/apimachinery/pkg/runtime"
+	kube_types "k8s.io/apimachinery/pkg/types"
+	kube_record "k8s.io/client-go/tools/record"
+	kube_ctrl "sigs.k8s.io/controller-runtime"
+	kube_client "sigs.k8s.io/controller-runtime/pkg/client"
+	kube_handler "sigs.k8s.io/controller-runtime/pkg/handler"
+	kube_reconile "sigs.k8s.io/controller-runtime/pkg/reconcile"
+	kube_source "sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/kumahq/kuma/pkg/core/resources/manager"
+	"github.com/kumahq/kuma/pkg/dns"
+	mesh_k8s "github.com/kumahq/kuma/pkg/plugins/resources/k8s/native/api/v1alpha1"
+	"github.com/kumahq/kuma/pkg/plugins/runtime/k8s/metadata"
+)
+
+// ConfigMapReconciler reconciles a ConfigMap object
+type ConfigMapReconciler struct {
+	kube_client.Client
+	kube_record.EventRecorder
+	Scheme          *kube_runtime.Scheme
+	Log             logr.Logger
+	ResourceManager manager.ResourceManager
+	IPAM            dns.IPAM
+	Persistence     *dns.MeshedPersistence
+	Resolver        dns.DNSResolver
+}
+
+func (r *ConfigMapReconciler) Reconcile(req kube_ctrl.Request) (kube_ctrl.Result, error) {
+	mesh, ok := dns.MeshedConfigKey(req.Name)
+	if !ok {
+		return kube_ctrl.Result{}, nil
+	}
+
+	if err := dns.CreateOrUpdateVIPConfig(r.Persistence, r.ResourceManager, r.Resolver, r.IPAM, mesh); err != nil {
+		return kube_ctrl.Result{}, err
+	}
+
+	return kube_ctrl.Result{}, nil
+}
```

Comment:
Nice, I like how simple this logic is now

Reply:
Yeah, easier to read now :)